src/core/thread/src_match_controller.cpp

```diff
@@ -31,6 +31,8 @@
  *   This file implements source address match controller.
  */
 
+#if OPENTHREAD_FTD
```

Comment:
Should we move this after the `#include "src_match_controller.hpp"` to give chance for config header to be included?